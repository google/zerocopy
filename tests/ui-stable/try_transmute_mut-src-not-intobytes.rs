../ui-nightly/try_transmute_mut-src-not-intobytes.rs