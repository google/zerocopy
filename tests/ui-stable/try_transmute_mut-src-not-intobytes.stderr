error[E0277]: the trait bound `NotZerocopy<AU16>: zerocopy::IntoBytes` is not satisfied
  --> tests/ui-stable/try_transmute_mut-src-not-intobytes.rs:19:52
   |
19 |     let src_not_into_bytes: Result<&mut AU16, _> = try_transmute_mut!(src);
   |                                                    ^^^^^^^^^^^^^^^^^^^^^^^ the trait `zerocopy::IntoBytes` is not implemented for `NotZerocopy<AU16>`
   |
   = note: Consider adding `#[derive(IntoBytes)]` to `NotZerocopy<AU16>`
   = help: the following other types implement trait `zerocopy::IntoBytes`:
             ()
             AU16
             AtomicBool
             AtomicI16
             AtomicI32
             AtomicI64
             AtomicI8
             AtomicIsize
           and $N others
note: required by a bound in `try_transmute_mut`
  --> src/util/macro_util.rs
   |
   | pub fn try_transmute_mut<Src, Dst>(src: &mut Src) -> Result<&mut Dst, ValidityError<&mut Src, Dst>>
   |        ----------------- required by a bound in this function
   | where
   |     Src: IntoBytes,
   |          ^^^^^^^^^ required by this bound in `try_transmute_mut`
   = note: this error originates in the macro `try_transmute_mut` (in Nightly builds, run with -Z macro-backtrace for more info)
